lighthouse-core/test/gather/driver-test.js

```diff
@@ -485,14 +678,19 @@ describe('.gotoURL', () => {
         },
       };
 
-      expect.assertions(2);
+      // 2 assertions in the catch block and the 1 implicit in `findListener`
+      expect.assertions(3);
 
       try {
         const loadPromise = driver.gotoURL(startUrl, loadOptions);
         await flushAllTimersAndMicrotasks();
+
+        const listener = driver.on.findListener('Security.securityStateChanged');
+        listener(insecureSecurityState);
+        await flushAllTimersAndMicrotasks();
         await loadPromise;
       } catch (err) {
-        expect(err.code).toEqual('INSECURE_DOCUMENT_REQUEST');
+        expect(err).toMatchObject({code: 'INSECURE_DOCUMENT_REQUEST'});
```

Comment:
old way seems slightly clearer to me (since it's just a single property)

Reply:
maybe, but it's a major PITA to debug when something goes wrong. this way will actually print the full error out so you see what was throwing.

compromise on `toHaveProperty`?